### PR TITLE
List Endpoint - Search Parameters and Pagination

### PR DESCRIPTION
**Pull request by LBHBReynolds-Carr**
- Created ClaimantQueryParam
- Implemented Search Parameters and Query
- Implemented pagination
- Edited database to remove foreign key dependency

### CODE REVIEW

File: AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs [hunk 1]
```diff
@@ -70,57 +68,226 @@ public void GetClaimantInformationByClaimIdAndPersonRefReturnsAddressDetails()
...
 
         [Test]
         public void GetAllClaimantsIfThereAreNoClaimantsReturnsAnEmptyList()
         {
-            _classUnderTest.GetAllClaimants().Should().BeEmpty();
+            _classUnderTest.GetAllClaimants(0, 20).Should().BeEmpty();
         }
 
         [Test]
         public void GetAllClaimantsIfThereAreClaimantsWillReturnThem()
```
Comment (elcorbs) [inline AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs]: This doesn't seem to be testing what the title suggests. The assertion is checking that the list is empty, which is the same as the test above.

File: AcademyResidentInformationApi/V1/UseCase/GetAllClaimantsUseCase.cs [hunk 2]
```diff
@@ -14,9 +15,12 @@ public GetAllClaimantsUseCase(IAcademyGateway academyGateway)
             _academyGateway = academyGateway;
         }
 
-        public ClaimantInformationList Execute()
+        public ClaimantInformationList Execute(ClaimantQueryParam cqp, int cursor, int limit)
         {
-            var claimants = _academyGateway.GetAllClaimants();
+            limit = limit < 10 ? 10 : limit;
+            limit = limit > 100 ? 100 : limit;
```
Comment (elcorbs) [inline AcademyResidentInformationApi/V1/UseCase/GetAllClaimantsUseCase.cs:21]: I might be missing them but I don't think are any tests around the limit functionality

**Review by elcorbs — COMMENTED** [review]
There are some files at the end of this PR that are related to what looks like a local session manager download -  I think they need to be removed

File: AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs (hunk 1 above)
Comment (LBHBReynolds-Carr) [reply AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs]: There's not been any major change to the test, but the method now requires a cursor and limit variable.

Comment (elcorbs) [reply AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs]: It looks like the test has changed from adding records to the database then checking they have all been returned to just checking that nothing has been returned.

File: AcademyResidentInformationApi/V1/UseCase/GetAllClaimantsUseCase.cs (hunk 2 above)
Comment (LBHBReynolds-Carr) [reply AcademyResidentInformationApi/V1/UseCase/GetAllClaimantsUseCase.cs:21]: I've added tests relating to the limit

File: AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs (hunk 1 above)
Comment (LBHBReynolds-Carr) [reply AcademyResidentInformationApi.Tests/V1/Gateways/AcademyGatewayTests.cs]: @elcorbs I've merged those tests together, as there will not be instances of housing benefit claimants who do not have an address, so demonstrating the claimant was returned could be combined with the test that they have been returned with an address.

**Review by elcorbs — APPROVED** [review]
Looks good 🎉

**Pull request merged**